packages/tools/viewer-alpha/src/viewer.ts

```diff
@@ -110,18 +160,121 @@ export class Viewer implements IDisposable {
         this._camera = new ArcRotateCamera("camera1", 0, 0, 1, Vector3.Zero(), this._details.scene);
         this._camera.attachControl();
         this._updateCamera(); // set default camera values
+        this._autoRotationBehavior = this._camera.getBehaviorByName("AutoRotation") as AutoRotationBehavior;
 
         // Load a default light, but ignore errors as the user might be immediately loading their own environment.
         this.loadEnvironmentAsync(undefined).catch(() => {});
 
         // TODO: render at least back ground. Maybe we can only run renderloop when a mesh is loaded. What to render until then?
-        this._engine.runRenderLoop(() => {
+        const render = () => {
             this._details.scene.render();
-        });
+            if (this.isAnimationPlaying) {
+                this.onAnimationProgressChanged.notifyObservers();
+                this._autoRotationBehavior.resetLastInteractionTime();
+            }
+        };
+
+        this._engine.runRenderLoop(render);
+        this._renderLoop = {
+            dispose: () => this._engine.stopRenderLoop(render),
+        };
 
         options?.onInitialized?.(this._details);
     }
 
+    /**
+     * The list of animation names for the currently loaded model.
+     */
+    public get animations(): readonly string[] {
+        return this._details.model?.animationGroups.map((group) => group.name) ?? [];

```

Comment:
should this create a new array every time ? shouldn t it be cached on load ?

Reply:
It could definitely be cached, it's just that it is called infrequently so personally preferring simpler less stateful code vs. very very finy performance impact. I don't feel strongly about it though, so I could switch it if you think it is much better to cache it.

Reply:
if it is used infrequently it is all good